lesson8/main.py

```diff
@@ -0,0 +1,73 @@
+import json
+from user import User
+
+
+def add_user(db, us):
+    db[us.name] = us.password
+
+
+def check_name(db):
+    while True:
+        username = input("Enter your username for a registration.\n")
+        if username in db.keys():
+            print("This username is already taken.")
+        elif 10 < len(username) or 2 > len(username):
+            print("The name must consists of less than 10 chars and more than 1.")
+        else:
+            return username
+
+
+def get_password():
+    password = input("Create your password:\n")
+    while input("Repeat your password:\n") != password:
+        pass
+    print("The registration is over. Now you can log-in.\n")
+    return password
+
+
+def login():
+    while True:
+        username = input("Please, enter your username:\n")
+        if username in data.keys():
+            print("Correct.")
+            break
+        else:
+            print("Authentication error. Try again.\n")
+    while True:
+        password = input("Please, enter your password:\n")
+        if (username, password) in data.items():
+            print(f"Hello, {username}!")
+            break
+        else:
+            print("Authentication error. Try again.\n")
+
+
+user1 = User("Alex", "11111")
```

Comment:
If I want to write data to the DB I dont need to define variable for this.. Better to do it using map function and initial dict.. but on this step its ok